src/main/java/org/openstreetmap/atlas/checks/validation/tag/DuplicateMapFeatureCheck.java

```diff
@@ -0,0 +1,491 @@
+            {
+                // Do Nothing
+            }
+        }
+
+        if (!duplicateFeaturesIds.isEmpty())
+        {
+            final Tuple<Set<String>, Map<String, String>> duplicateFeatures = Tuple
+                    .createTuple(duplicateFeaturesIds, duplicateFeaturesTags);
+
+            return Optional.of(duplicateFeatures);
+        }
+
+        return Optional.empty();
+    }
+
+    private String getStringForList(final Set<String> list)
+    {
+        return list.toString().replace("[", "").replace("]", "");
+    }
+
+    /**
+     * Checks if two set tags have the key but different value
+     *
+     * @param firstTags
+     *            the first Osm tags to check
+     * @param secondTags
+     *            the second Osm tags to check
+     * @param key
+     *            the key to check
+     * @return true if two set tags have the key but different value.
+     */
+    private boolean hasDifferentValueOfKey(final Map<String, String> firstTags,
+            final Map<String, String> secondTags, final String key)
+    {
+        return firstTags.get(key) != null && secondTags.get(key) != null
+                && !firstTags.get(key).equals(secondTags.get(key));
+    }
+
+    /**
+     * Checks if two set tags have only building tag without name for features which should
+     * represent once
+     *
+     * @param firstTags
+     *            the first Osm tags to check
+     * @param secondTags
+     *            the second Osm tags to check
+     * @param featuresTaggedTwice
+     *            the features tags in both sets of tags for features which should represent once
+     * @return true if two set tags have only building tag without name for features which should
+     *         represent once.
+     */
+    private boolean hasOnlyBuildingWithoutName(final Map<String, String> firstTags,
+            final Map<String, String> secondTags, final Map<String, String> featuresTaggedTwice)
+    {
+        return featuresTaggedTwice.keySet().contains(BuildingTag.KEY)
+                && featuresTaggedTwice.keySet().size() == 1
+                && (firstTags.get(NameTag.KEY) == null || secondTags.get(NameTag.KEY) == null);
+    }
+
+    /**
+     * Checks if the {@link AtlasEntity} is a member of the {@link Relation}.
+     *
+     * @param relation
+     *            {@link Relation} to check
+     * @param entity
+     *            {@link AtlasEntity} to check
+     * @return true if the AtlasEntity is a member of the Relation
+     */
+    private boolean isRelationMember(final Relation relation, final AtlasEntity entity)
+    {
+        final List<RelationMember> members = relation.members().stream()
+                .collect(Collectors.toList());
+
+        for (final RelationMember relationMember : members)
+        {
+            if (relationMember.getEntity().getOsmIdentifier() == entity.getOsmIdentifier())
+            {
+                return true;
+            }
+        }
+        return false;
```

Comment:
You need to check that the member and entity are of the same type, as identifiers are not unique across entity types. This can also be shortened using a stream. 
```suggestion
        final List<RelationMember> members = relation.members().stream().anyMatch(member -> member.getEntity().getType().equals(entity.getType()) && member.getEntity().getOsmIdentifier() == entity.getOsmIdentifier()));
```